libcst/_nodes/statement.py

```diff
@@ -1472,6 +1475,40 @@ def _codegen_impl(self, state: CodegenState) -> None:
         self.trailing_whitespace._codegen(state)
 
 
+def get_docstring_impl(
+    body: Union[BaseSuite, Sequence[Union[SimpleStatementLine, BaseCompoundStatement]]],
+    clean: bool,
+) -> Optional[str]:
+    """
+    Implementation Reference:
+    - :func:`ast.get_docstring` https://docs.python.org/3/library/ast.html#ast.get_docstring
+    and https://github.com/python/cpython/blob/89aa4694fc8c6d190325ef8ed6ce6a6b8efb3e50/Lib/ast.py#L254
+    - PEP 257 https://www.python.org/dev/peps/pep-0257/
+    """
+    if isinstance(body, Sequence):
+        if body:
+            expr = body[0]
+        else:
+            return None
+    else:
+        expr = body
+    while isinstance(expr, (BaseSuite, SimpleStatementLine)):
+        if len(expr.body) == 0:
+            return None
+        expr = expr.body[0]
+    if not isinstance(expr, Expr):
+        return None
+    val = expr.value
+    if isinstance(val, (SimpleString, ConcatenatedString)):
+        evaluated_value = val.evaluated_value
+    else:
+        return None
```

Comment:
Technically grouping parens are allowed, I'm not sure this handles that case.

```
>>> def f():
...     ("doc")
... 
>>> f.__doc__
'doc'
```

Reply:
Yes, it works since the parentheses are formatting data which parsed as attribute of the SimpleString. See the paste for more detail:

```
SimpleStatementLine(
                body=[
                    Expr(
                        value=SimpleString(
                            value='"doc"',
                            lpar=[
                                LeftParen(
                                    whitespace_after=SimpleWhitespace(
                                        value='',
                                    ),
                                ),
                            ],
                            rpar=[
                                RightParen(
                                    whitespace_before=SimpleWhitespace(
                                        value='',
                                    ),
                                ),
                            ],
                        ),
                        semicolon=MaybeSentinel.DEFAULT,
                    ),
                ],
```


Also add it as a test case.